bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/Page.java

```diff
@@ -467,4 +469,27 @@ default List<String> getRobotsTags() {
      */
     default boolean isDataLayerClientlibIncluded() {return true;}
 
+    /**
+     * Returns the name of the data layer.
+     *
+     * {@code true} if the data layer client library should be included.
+     * @since om.adobe.cq.wcm.core.components.models 12.24.0
+     */
+    default String getDataLayerName() { return DATALAYER_OBJECT_NAME_ADOBE; }
+
+    /**
+     * Checks if gtag() should be used to push to the data layer.
+     *
+     * {@code true} if gtag() should be used.
+     * @since om.adobe.cq.wcm.core.components.models 12.24.0
+     */
+    default boolean isDataLayerUseGtag() { return false; };
+
+    /**
+     * Checks if the ACDL library should be included.
+     *
+     * {@code true} if ACDL library should be used.
+     * @since om.adobe.cq.wcm.core.components.models 12.24.0
+     */
+    default boolean isDataLayerSkipAcdlInclude() { return false; };
```

Comment:
We already have `isDataLayerClientlibIncluded` that serves the same purpose (to control if the ACDL scripts should be included).

Reply:
hi @vladbailescu  

unfortunately isDataLayerClienlibIncluded removes the datalayer.js file not just the ACDL.  This may be a bug: when the flag is set to true there is an odd state - some component datalayer interaction happens as before (accordian, text etc), but the components that are caught in a loop in datalayer.js are no longer pushed to the datalayer.  In short we get a semi-enabled state.

This is the reason: 

assets: {
                js: [
                    "src/scripts/datalayer/v1/polyfill.js",
                    "node_modules/@adobe/adobe-client-data-layer/dist/adobe-client-data-layer.min.js",
                    "src/scripts/datalayer/v1/datalayer.js"
                ]
            }

 For the GDL I need to keep datalayer.js.  This has led to the ugly approach of "take DL away and put part of it back again"

Perhaps it would be better to change the existing functionality so that only the ACDL code is removed by isDataLayerClientlibIncluded?  If so I can include that here and it would allow me to make the whole thing cleaner.